### ISSUE TITLE
Yield fuction to work with multiple signal at once.

### ISSUE THREAD

**Issue by irfan-hossain-bhuiyan** [godotengine/godot-proposals#2622]
### Describe the project you are working on
creating A simple UI

### Describe the problem or limitation you are having in your project
In my main project I face the problem like this
```
func _ready():
	open_popup_menu()
	#open_popup_menu() opens a custom window and there are two button
	#buttonA and buttonB
	#then I want to check if any the button is pressed than run somefunc()
	#if yield could support multiple signal at once than i could just
	yield([buttonA,"pressed"] or [buttonB,"pressed"])
        somefunc()
        #there is or in yield function because there can be something 
        #like yield([buttonA,"pressed"] and [buttonB,"pressed"]) when 
        #need to run somefunc() after both of the button are pressed(not any current order.)
	

```

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Yield can't work with multiple signal.
### If this enhancement will not be used often, can it be worked around with a few lines of script?
As the yield couldn't support multiple signal,
1.I have to create other node name nodeA
2.connect both of the button's signal to nodeA to funcA,funcB
3.In funcA and funcB I will emit other custom signal singalC
4.than yield from that signal.
### Is there a reason why this should be core and not an add-on in the asset library?
It would be a feature of gdscript,So addon in the asset library maybe not what we want.

**Comment by pycbouh** [comment]
`yield`, and `await` in 4.0, exist for very simple and non branching flows. If your flow is branching you are expected to fully utilize signals and `connect` method.

I am also not sure why you’d need another node here. You can connect everything through the same script as the one that golds your `other` method.

**Comment by irfan-hossain-bhuiyan** [comment]
Oh sorry,I actually did it to maintain my code,but it's not needed,Should I edit it?

**Comment by pycbouh** [comment]
Sure, you can correct your proposal.

**Comment by dalexeev** [comment]
In most cases, you can add a signal that is emitted when the last of the desired signals is emitted. See [example](https://github.com/godotengine/godot-proposals/issues/2518#issuecomment-812440622).

**Comment by irfan-hossain-bhuiyan** [comment]
Yeah,But isn't it much faster if the feature is implemented.It will decrease complexity in the code,You don't have to create a new signal for it.Also things get more complex if you want to do 
Yield("singalA" and "signalb")
means the yield will trigger after both of the signals are emitted.
I don't know how much unstable the engine would be,if the feature is implemented,but I can find a lot of use case for it.

**Comment by pycbouh** [comment]
The syntax will unlikely change to support multiple parameters, not because it will somehow make the engine more "unstable", but because there is a whole system for branching behaviors using signals, and there is no point trying to cram it all into a concise one line statement.

The best that can happen is something analogous to `Promise.all`, `Promise.any`, `Promise.race` from JavaScript. But you still better master signals and spread your logic across several methods. It will result in cleaner design.

**Comment by me2beats** [comment]
Personally I like the idea, it even could be a keyword like `when_all` (as well as `when_any`). 

This is a fairly common task when you want to wait for one of several actions, or all of the listed actions to be completed.
The problem with `connect()` is that it is rather verbose.

However, I take into account that this is possibly a tricky feature to implement in gdscript.
